src/components/world/Actor.tsx

```diff
@@ -0,0 +1,130 @@
+import { getSettings } from "store/features/settings/settingsState";
+import { RootState } from "store/configureStore";
+import { Tool } from "store/features/editor/editorState";
+import { Palette } from "store/features/entities/entitiesTypes";
+
+interface ActorProps {
+  id: string;
+  sceneId: string;
+  palettes: Palette[];
+  editable: boolean;
+}
+
+const Actor = (props: ActorProps) => {
+  const dispatch = useDispatch();
+
+  const dragActorStart = (payload: { actorId: string; sceneId: string }) =>
+    dispatch(editorActions.dragActorStart(
+      payload
+    ));
+
+  const dragActorStop = React.useCallback(() =>
+    dispatch(editorActions.dragActorStop()),
+  [dispatch]
+  );
+
+  const setTool = (payload: { tool: Tool }) =>
+    dispatch(editorActions.setTool(payload));
+
+  const { type: editorType, entityId, scene: sceneId } = useSelector((state: RootState) => state.editor);
+
+  const actor = useSelector((state: RootState) => actorSelectors.selectById(state, props.id));
+
+  const selected =
+    editorType === "actor" &&
+    sceneId === props.sceneId &&
+    entityId === props.id;
+  const showSprite = useSelector((state: RootState) => state.editor.zoom > 80);
+  const settings = useSelector((state: RootState) => getSettings(state));
+  const palettesLookup = useSelector((state: RootState) => paletteSelectors.selectEntities(state));
+
+  const palette = React.useMemo(() => {
+    if (!actor) return null;
+    const gbcEnabled = settings.customColorsEnabled;
+
+    return gbcEnabled
+      ? getCachedObject(
+        palettesLookup[actor.paletteId] ||
+        palettesLookup[settings.defaultSpritePaletteId]
+      )
+      : DMG_PALETTE as Palette;
+  }, [actor, palettesLookup, settings]);
+
+  /**
+   * This section of code is meant to setup `onMouseUp` and make sure that it
+   * simultaneously handles if/when "dragActorStop" dep is updated, but also
+   * to make sure that there are no memory leaks when that is done due to
+   * removeEventListener not being passed the same reference of function
+   *
+   * I wish this code could be cleaner, but doing the 1:1 migration to TS/FC,
+   * tis not
+   */
+  const isMouseListening = React.useRef(false);
+
+  const onMouseUp = React.useRef(() => {})
+
+  React.useEffect(() => {
+    const oldFn = onMouseUp.current;
+    onMouseUp.current = () => {
+      dragActorStop();
+      window.removeEventListener("mouseup", onMouseUp.current);
+      isMouseListening.current = false;
+    };
+    if (isMouseListening.current) {
+      window.removeEventListener("mouseup", oldFn);
+      window.addEventListener("mouseup", oldFn);
+    }
+  }, [dragActorStop])
+
+  const onMouseDown: MouseEventHandler = (e) => {
+    if (props.editable && e.nativeEvent.which !== MIDDLE_MOUSE && actor) {
+      dragActorStart({ sceneId, actorId: actor.id });
+      setTool({ tool: "select" });
+      window.addEventListener("mouseup", onMouseUp.current);
+      isMouseListening.current = true;
+    }
+  };
```

Comment:
This is a particularly tricky bit of code. Ideally, we should probably look into refactoring how this is done long-term to be a bit more clean. However, this code is as direct of a port of the class-based functionality as I could manage. 

I did a tiny bit of extra legwork to make sure that we're doing cleanups and re-assigning `onMouseUp` when `dragActorStop` is changed. This should at least make this code stable enough to use in production until we can rethink how to handle this logic